### PR TITLE
Add/wordpres core tests

### PR DESCRIPTION
**Pull request by pschoffer**
## Description

This PR aims to include WP core tests integrated to our build pipeline.

## Checklist

Please make sure the items below have been covered before requesting a review:

- [ ] This change works and has been tested locally (or has an appropriate fallback).
- [ ] This change works and has been tested on a Go sandbox.
- [ ] This change has relevant unit tests (if applicable).
- [ ] This change has relevant documentation additions / updates (if applicable).

## Steps to Test

Check CircleCI build pipelines.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -2,14 +2,47 @@ workflows:
...
       - php73-build-multisite
       - php73-build-singlesite
+      - php74-lint
       - php74-build-multisite
       - php74-build-singlesite
       - php74-build-multisite-nightly
       - php74-build-singlesite-nightly
 
-version: 2
+version: 2.1
```
Comment (pschoffer) [inline .circleci/config.yml:15]: I bumped the version so that I can use parameters

File: .circleci/config.yml [hunk 2]
```diff
@@ -35,9 +68,6 @@ job-references:
       git submodule update --init --recursive
 
   php_job: &php_job
-    environment:
```
Comment (pschoffer) [inline .circleci/config.yml:38]: Turns out these env variables were ignored. As when this block gets included in a job, job's environment section overwrites (not merges) with these. That is why I moved to use of pipeline parameters above instead.

File: .circleci/config.yml [hunk 3]
```diff
@@ -46,19 +76,61 @@ job-references:
...
+            npm run grunt build
+            export WP_TESTS_DIR=<< pipeline.parameters.wp_core_develop_dir >>/tests/phpunit
+            echo "test EN"
+            echo $WP_TESTS_DIR
+            ~/project/vendor/bin/phpunit --log-junit << pipeline.parameters.phpunit_report_file >> || true
+            ~/project/vendor/bin/phpunit --bootstrap ~/project/tests/bootstrap.php --log-junit << pipeline.parameters.phpunit_report_file_including_mu_plugins >> || true
+            diff <(cat << pipeline.parameters.phpunit_report_file >> | grep -o '<testsuite name="".*time=') <(cat << pipeline.parameters.phpunit_report_file_including_mu_plugins >> | grep -o '<testsuite name="".*time=')
+
+
+  lint_job: &lint_job
```
Comment (pschoffer) [inline .circleci/config.yml:106]: I have moved out lining to a separate more lightweight = faster job, which is independent to the core tests work

File: bin/install-wp-develop.sh [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+if [ ! -f wp-tests-config.php ]; then
+	cp wp-tests-config-sample.php wp-tests-config.php
+	sed $IOPTION "s/youremptytestdbnamehere/$DB_NAME/" wp-tests-config.php
+	sed $IOPTION "s/yourusernamehere/$DB_USER/" wp-tests-config.php
+	sed $IOPTION "s/yourpasswordhere/$DB_PASS/" wp-tests-config.php
+	sed $IOPTION "s|localhost|${DB_HOST}|" wp-tests-config.php
+fi
+
+
+install_db() {
```
Comment (pschoffer) [inline bin/install-wp-develop.sh:69]: This functuon is shared with `install-wp-tests.sh` and should be extracted and unified.

Comment (pschoffer) [reply bin/install-wp-develop.sh]: Done

File: .circleci/config.yml [hunk 5]
```diff
@@ -2,14 +2,41 @@ workflows:
   version: 2
   main:
     jobs:
+      - php73-lint
```
Comment (pschoffer) [inline .circleci/config.yml:5]: I am not sure if it makes sense to have both php73-lint and php74-lint?

**Comment by pschoffer** [comment]
I figured it might be good idea to get this merge already, and then do the whole hooking up mu-plugins to core-tests in separate PR.

**Review by brandon-m-skinner — APPROVED** [review]

**Comment by brandon-m-skinner** [comment]
Merged so you can hook up mu-plugins at your leisure.

**Pull request merged**